src/plugins/intel_gpu/src/graph/network.cpp

```diff
@@ -321,6 +323,101 @@ network::network(program::ptr program, uint16_t stream_id)
 network::network(program::ptr program, stream::ptr stream, uint16_t stream_id)
     : network(program, stream, false, stream_id == 0) {}
 
+network::network(cldnn::BinaryInputBuffer& ib, stream::ptr stream, engine& engine, uint16_t stream_id)
+    : _program(nullptr)
+    , _engine(ib.get_engine())
+    , _stream(stream)
+    , _memory_pool(new memory_pool(engine))
+    , _internal(false)
+    , _is_primary_stream(false)
+    , _reset_arguments(true) {
+    net_id += 1;
+
+    uint32_t prog_id;
+    std::vector<std::string> batch_header_str;
+    ib >> prog_id;
+    ib >> batch_header_str;
+    kernels_cache kernels_cache(get_engine(), prog_id, batch_header_str);
+    ib >> kernels_cache;
+
+    int num_data_nodes;
+    ib >> num_data_nodes;
+
+    _memory_pool->clear_pool_for_network(net_id);
+
+    for (int i = 0; i < num_data_nodes; ++i) {
+        std::string type;
+        std::string _primitive_id;
+        ib >> type >> _primitive_id;
+        std::shared_ptr<cldnn::primitive_inst> new_primitive_inst = cldnn::get_type_id(type)->create_instance(*this);
```

Comment:
Why do we need to separate data nodes and other node types here?

Reply:
In the case of deserialization, output memory is allocated whenever each `primitive_inst` is restored. At this time, in the case of `primitive_inst` that does not allocate memory on its own (`_mem_allocated` is false), the memory address of another `primitive_inst` is used. In the case of non-data types, there is no problem if they are restored in the order of `_exec_order`. But, since data types are not in `_exec_order`, memory addresses may not be known unless allocated in advance.